src/uct/ib/base/ib_md.c

```diff
@@ -553,22 +554,41 @@ ucs_status_t uct_ib_dereg_mrs(struct ibv_mr **mrs, size_t mr_num)
     return status;
 }
 
-static ucs_status_t uct_ib_memh_dereg(uct_ib_md_t *md, uct_ib_mem_t *memh)
+static ucs_status_t uct_ib_memh_dereg_key(uct_ib_md_t *md, uct_ib_mem_t *memh)
 {
-    ucs_status_t s1, s2;
-
     if (memh->flags & UCT_IB_MEM_MULTITHREADED) {
         return md->ops->dereg_multithreaded(md, memh);
+    } else {
+        return md->ops->dereg_key(md, memh);
     }
+}
+
+static ucs_status_t uct_ib_memh_dereg(uct_ib_md_t *md, uct_ib_mem_t *memh)
+{
+    ucs_status_t s, status = UCS_OK;
 
-    s2 = UCS_OK;
     if (memh->flags & UCT_IB_MEM_FLAG_ATOMIC_MR) {
-        s2 = md->ops->dereg_atomic_key(md, memh);
+        s = md->ops->dereg_atomic_key(md, memh);
         memh->flags &= ~UCT_IB_MEM_FLAG_ATOMIC_MR;
+        if (s != UCS_OK) {
+            status = s;
+        }
+    }
+
+    if (memh->flags & UCT_IB_MEM_FLAG_RELAXED_ORDERING) {
+        s = uct_ib_memh_dereg_key(md, memh);
+        memh->flags &= ~UCT_IB_MEM_FLAG_RELAXED_ORDERING;
```

Comment:
this looks weird, UCT_IB_MEM_FLAG_RELAXED_ORDERING means we need to call uct_ib_memh_dereg_key twice?